File: algorithms/sort/simple/selection/Selection.java
import java.util.Arrays;

/** Check README to find description */
public class Selection {
    public void sort(int[] a) {
        for(int i = 0; i < a.length - 1; i++) {
            int ind = getIndOfSmallest(a, i, a.length);
            swap(a, i, ind);
        }
    }

    private int getIndOfSmallest(int[] a, int start, int end) {
        int smallestInd = start;
        for(int i = start + 1; i < end; i++) {
            if(a[i] < a[smallestInd]) {
                smallestInd = i;
            }
        }
        return smallestInd;
    }

    private void swap(int[] a, int ind1, int ind2) {
        if(ind1 == ind2) return;
        int tmp = a[ind1];
        a[ind1] = a[ind2];
        a[ind2] = tmp;
    }

    public static void main(String[] args) {
        int[] array = new int[] {10, 2, 11, 31, 60, 90, 81,
            85, 75, 21, 13, 32, 42, 90, 1};
        System.out.println(Arrays.toString(array));
        new Selection().sort(array);
        System.out.println(Arrays.toString(array));
    }
}

File: algorithms/sort/simple/selection/Selection.kt
import java.util.Arrays;

/** Check README to find description */
class Selection {
    fun sort(a: IntArray) {
        for (i in 0 until a.size - 1) {
            var minPos = i
            for (j in minPos + 1 until a.size) {
                if (a[minPos] > a[j]) {
                    minPos = j
                }
            }
            if (minPos != i) {
                val tmp = a[minPos]
                a[minPos] = a[i]
                a[i] = tmp
            }
        }
    }
}

fun main(args: Array<String>) {
    val array = intArrayOf(10, 2, 11, 31, 60, 90, 81,
        85, 75, 21, 13, 32, 42, 90, 1)
    System.out.println(Arrays.toString(array))
    Selection().sort(array)
    System.out.println(Arrays.toString(array))
}
